File: lib/features/storekeeper/domain/models/catering_reconciliation.dart
/// Domain model for a reconciliation record.
class CateringReconciliation {
  final String id;
  final String companyId;
  final String issuanceId;
  final String storekeeperId;
  final int totalIssuedValuePaisa;
  final int totalReturnedValuePaisa;
  final int totalSoldValuePaisa;
  final int variancePaisa;
  final String status; // draft | confirmed | disputed
  final String? notes;
  final DateTime? reconciledAt;
  final DateTime createdAt;

  const CateringReconciliation({
    required this.id,
    required this.companyId,
    required this.issuanceId,
    required this.storekeeperId,
    this.totalIssuedValuePaisa = 0,
    this.totalReturnedValuePaisa = 0,
    this.totalSoldValuePaisa = 0,
    this.variancePaisa = 0,
    this.status = 'draft',
    this.notes,
    this.reconciledAt,
    required this.createdAt,
  });

  bool get isDraft => status == 'draft';
  bool get isConfirmed => status == 'confirmed';

  factory CateringReconciliation.fromJson(Map<String, dynamic> json) {
    return CateringReconciliation(
      id: json['id'] ?? '',
      companyId: json['company_id'] ?? '',
      issuanceId: json['issuance_id'] ?? '',
      storekeeperId: json['storekeeper_id'] ?? '',
      totalIssuedValuePaisa: _parseInt(json['total_issued_value_paisa']),
      totalReturnedValuePaisa: _parseInt(json['total_returned_value_paisa']),
      totalSoldValuePaisa: _parseInt(json['total_sold_value_paisa']),
      variancePaisa: _parseInt(json['variance_paisa']),
      status: json['status'] ?? 'draft',
      notes: json['notes'],
      reconciledAt: json['reconciled_at'] != null ? DateTime.tryParse(json['reconciled_at']) : null,
      createdAt: DateTime.tryParse(json['created_at'] ?? '') ?? DateTime.now(),
    );
  }

  static int _parseInt(dynamic v) {
    if (v is int) return v;
    return int.tryParse(v?.toString() ?? '0') ?? 0;
  }
}

File: backend/app/Models/CateringReconciliation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class CateringReconciliation extends Model
{
    protected $table = 'catering_reconciliations';

    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        'company_id',
        'issuance_id',
        'storekeeper_id',
        'total_issued_value_paisa',
        'total_returned_value_paisa',
        'total_sold_value_paisa',
        'total_wasted_value_paisa',
        'total_staff_value_paisa',
        'total_complimentary_value_paisa',
        'variance_paisa',
        'status',
        'notes',
        'reconciled_at',
    ];

    protected $casts = [
        'total_issued_value_paisa'   => 'integer',
        'total_returned_value_paisa' => 'integer',
        'total_sold_value_paisa'     => 'integer',
        'total_wasted_value_paisa'       => 'integer',
        'total_staff_value_paisa'        => 'integer',
        'total_complimentary_value_paisa' => 'integer',
        'variance_paisa'             => 'integer',
        'reconciled_at'              => 'datetime',
    ];

    protected static function boot()
    {
        parent::boot();
        static::creating(function (self $model) {
            if (empty($model->id)) {
                $model->id = (string) Str::orderedUuid();
            }
        });
    }

    public function company()
    {
        return $this->belongsTo(Company::class);
    }

    public function issuance()
    {
        return $this->belongsTo(CateringIssuance::class, 'issuance_id');
    }

    public function storekeeper()
    {
        return $this->belongsTo(StoreKeeper::class, 'storekeeper_id');
    }

    public function scopeForCompany($query, string $companyId)
    {
        return $query->where('company_id', $companyId);
    }

    public function scopeDraft($query)
    {
        return $query->where('status', 'draft');
    }

    public function confirm(): void
    {
        $this->update(['status' => 'confirmed', 'reconciled_at' => now()]);
    }

    public function dispute(): void
    {
        $this->update(['status' => 'disputed']);
    }
}
